namer/namer.cc

```diff
@@ -346,10 +339,9 @@ class NameInserter {
         klass->symbol.data(ctx)->addLoc(ctx, klass->declLoc);
         klass->symbol.data(ctx)->singletonClass(ctx); // force singleton class into existence
 
-        auto toRemove = remove_if(klass->rhs.begin(), klass->rhs.end(), [&](unique_ptr<ast::Expression> &line) {
-            return handleNamerRewriter(ctx, klass, line);
-        });
-        klass->rhs.erase(toRemove, klass->rhs.end());
+        for (auto &exp : klass->rhs) {
+            handleNamerRewriter(ctx, klass, exp);
+        }
 
         if (!klass->ancestors.empty()) {
```

Comment:
One thing that would be tempting to do with the ancestors also still in the tree is to have people think that they should treewalk to find `include` / `extend` when we'd prefer that they walk over the `ancestors` list.

If I had to guess, this is the reason that it's implemented the way it is right now: because @DarkDimius wanted there to only be one way to do something that involved ancestors.

Reply:
Yeah, I bet that was it. I think we're far enough along in the coding that people won't pull from this as their source of truth